code/game/turfs/simulated/lava.dm

```diff
@@ -95,6 +95,6 @@
 
 // Tells AI mobs to not suicide by pathing into lava if it would hurt them.
 /turf/simulated/floor/lava/is_safe_to_enter(mob/living/L)
-	if(!is_safe() && !L.hovering)
+	if(!is_safe() && !L.hovering && !(L.is_incorporeal()))

```

Comment:
This won't actually make them immune to lava. You'd wanna do that down in burn_stuff()

Reply:
Okay, I kind of knew that but when actually doing changes I was for some reason 100% certain proc i was modifying was is_safe... despite the fact that is_safe is referenced in that very check. Either way. Fixed.